prime-angular-frontend/src/app/modules/enrolment/pages/pharmanet-enrolment-summary/pharmanet-enrolment-summary.component.ts

```diff
@@ -137,62 +140,25 @@ export class PharmanetEnrolmentSummaryComponent extends BaseEnrolmentPage implem
     }
   }
 
-  public isEmailVisible(careSettingCode: number): boolean {
-    switch (careSettingCode) {
-      case this.CareSettingEnum.PRIVATE_COMMUNITY_HEALTH_PRACTICE: {
-        return this.showCommunityHealth;
-      }
-      case this.CareSettingEnum.COMMUNITY_PHARMACIST: {
-        return this.showPharmacist;
-      }
-      case this.CareSettingEnum.HEALTH_AUTHORITY: {
-        return this.showHealthAuthority;
-      }
-      case this.CareSettingEnum.DEVICE_PROVIDER: {
-        return this.showDeviceProvider;
-      }
-      default: {
-        return false;
-      }
-    }
-  }
-
   public getTokenUrl(tokenId: string): string {
     return `${this.config.loginRedirectUrl}/provisioner-access/${tokenId}`;
   }
 
-  public sendProvisionerAccessLinkTo(careSettingCode: number) {
-    let formControl: FormControl;
-
-    switch (careSettingCode) {
-      case this.CareSettingEnum.PRIVATE_COMMUNITY_HEALTH_PRACTICE: {
-        formControl = this.communityHealthEmails;
-        break;
-      }
-      case this.CareSettingEnum.COMMUNITY_PHARMACIST: {
-        formControl = this.pharmacistEmails;
-        break;
-      }
-      case this.CareSettingEnum.HEALTH_AUTHORITY: {
-        formControl = this.healthAuthorityEmails;
-        break;
+  public sendProvisionerAccessLinkTo() {
+    let valid = true;
+    this.careSettingConfigs.forEach((config) => {
+      valid = valid && config.formArray.valid;
+      if (!config.formArray.valid) {
+        config.formArray.markAllAsTouched();
```

Comment:
Why is this line needed?

Reply:
Maybe a better way to do it, but it is needed. It checks if all email address form validation is valid. If it's not, it will trigger the UI validation by .markAllAsTouched().

Reply:
IMO, worth adding as a comment